app/services/record_declaration.rb

```diff
@@ -131,20 +131,22 @@ def validate_milestone_exists
 
   def participant_declaration
     @participant_declaration ||= find_participant_declaration.tap do |pd|
-      pd.update!(uplift_flags.merge(cohort:))
+      pd.update!(uplift_flags)
     end
   end
 
   def find_participant_declaration
-    participant_profile
+    existing_declaration = participant_profile
       .participant_declarations
       .submitted
       .or(
         participant_profile
           .participant_declarations
           .billable,
       )
-      .create_or_find_by!(declaration_parameters)
+      .find_by(declaration_parameters)
+
+    existing_declaration || participant_profile.participant_declarations.create!(declaration_parameters.merge(cohort:))
```

Comment:
do you mind adding a spec for this to check that it doesn't try to create a new declaration if there is an existing one in another cohort?